conan/tools/ros/rosenv.py

```diff
@@ -0,0 +1,38 @@
+import os
+from conan.api.output import Color
+from conan.tools.files import save
+
+
+class ROSEnv(object):
+    """
+    Generator to serve as integration for Robot Operating System 2 development workspaces.
+    It generates a conanrosenv.bash file that when sources sets variables so the Conan
+    dependencies are found by CMake and the run environment is also set.
+
+    IMPORTANT: This generator should be used together with CMakeDeps and CMakeToolchain generators.
+    """
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.filename = "conanrosenv.bash"
```

Comment:
why not ``.sh`` like others?

Reply:
ROS users are used to "source" bash files as this is what ament/colcon generates to activate their environment.
For example:
```
$ source /opt/ros/humble/setup.bash  --> to activate ros environment
$ source install/setup.bash --> to activate the environment to run apps/nodes built in a worksapce
```
https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Configuring-ROS2-Environment.html#source-the-setup-files

https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Creating-Your-First-ROS2-Package.html#source-the-setup-file